### DISCUSSION TITLE
Memory Leak within `load_ssl_context_verify` when using openai-python

### DISCUSSION THREAD

**Discussion by vasudua** [encode/httpx#3272] [Potential Issue]
## Description
`langchain` initialises an `openai-python` instance which in turn defines a default httpx client something like [here](https://github.com/openai/openai-python/blob/798c6cb8032f4568d0b23f762d479d68be62ce3e/src/openai/_base_client.py#L1420) when a custom http_client is not passed as an argument to the class initialization. And this httpx client defined continues to hold memory even after the parent object is destroyed.

The place where the memory is being held up is `load_ssl_context_verify`, memray results follow.
<img width="1167" alt="Screenshot 2024-08-19 at 12 29 04 PM" src="https://github.com/user-attachments/assets/aaa7e28b-f1aa-455d-b7ae-0387f5340a19">
As shown in the above screenshot, it is holding ~88 MBs of memory. This memory keeps growing with each new instance of the above classes created and never gets released even when the parent objects of the above mentioned libs are destroyed.

## Expected Behaviour
The memory should be released once the parent instances are destroyed.

**Comment by giorgiodig** [comment]
seeing the same behaviour here...

**Comment by giorgiodig** [comment]
@vasudua did you figure things out? which version were you using?
